README.md

```diff
@@ -8,35 +8,467 @@ The Pinecone Java client lets JVM applications interact with Pinecone services.
+
+        OkHttpClient httpClient = builder.build();
+
+        Pinecone pinecone = new Pinecone.Builder("PINECONE_API_KEY").build();
+    }
+}
+```
+
+# Indexes
+Operations related to the building and managing of Pinecone indexes are called [control plane](https://docs.pinecone.io/reference/api/introduction#control-plane) operations.
+
+## Create Index
+You can use the Java SDK to create two types of indexes: [serverless indexes](https://docs.pinecone.io/guides/indexes/understanding-indexes#serverless-indexes) (recommended for most use cases) and 
+[pod-based indexes](https://docs.pinecone.io/guides/indexes/understanding-indexes#pod-based-indexes) (recommended for high-throughput use cases).
+
+### Create a serverless index
+
+> [!WARNING]  
+> Serverless indexes are in **public preview** and are available only on AWS in the
+> `us-east-1` and `us-west-2` regions. Check the [current limitations](https://docs.pinecone.io/docs/limits#serverless-index-limitations) and test thoroughly before using it in production.
+
+```java
+import io.pinecone.clients.Pinecone;
+import org.openapitools.client.model.*;
+
+public class CreateServerlessIndexExample {
+    public static void main(String[] args) {
+        Pinecone pinecone = new Pinecone.Builder("PINECONE_API_KEY").build();
+        
+        String indexName = "example-index";
+        String similarityMetric = "cosine";
+        Integer dimension = 1538;
+        String cloud = "aws";
+        String region = "us-west-2";
+
+        pinecone.createServerlessIndex(indexName, similarityMetric, dimension, cloud, region);
+    }
+}
+```
+
+### Create a pod index
+The following is a minimal example of creating a pod-based index. For all the possible configuration options, see 
+`main/java/io/pinecone/clients/Pinecone.java`.
+
+```java
+import io.pinecone.clients.Pinecone;
+import org.openapitools.client.model.*;
+
+public class CreatePodIndexExample {
+    public static void main(String[] args) {
+        Pinecone pinecone = new Pinecone.Builder("PINECONE_API_KEY").build();
+        
+        String indexName = "example-index";
+        String similarityMetric = "cosine"; // Optional; defaults to cosine similarity
+        Integer dimension = 1538;
+        String environment = "us-east-1-aws";
+        String podType = "p1.x1";
+        
+        pinecone.createPodsIndex(indexName, similarityMetric, dimension, environment, podType);
+    }
+}
+```
+
+## List indexes
+
+The following example returns all indexes (and their corresponding metadata) in your project.
+
+```java
+import io.pinecone.clients.Pinecone;
+import org.openapitools.client.model.*;
+
+public class ListIndexExample {
+    public static void main(String[] args) {
+        Pinecone pinecone = new Pinecone.Builder("PINECONE_API_KEY").build();
+        
+        IndexList indexesInYourProject = pinecone.listIndexes();
+        
+        System.out.println(indexesInYourProject);
+    }
```

Comment:
When these methods return something I like how you assigned it so the return type is visible. 

We can probably exclude the System.out.println statements

Reply:
I followed this pattern as well in my PR. https://github.com/pinecone-io/pinecone-java-client/pull/109